src/librustc_lexer/src/unescape.rs

```diff
@@ -124,12 +122,7 @@ impl Mode {
 fn scan_escape(first_char: char, chars: &mut Chars<'_>, mode: Mode) -> Result<char, EscapeError> {
     if first_char != '\\' {
         return match first_char {
-            '\t' | '\n' => Err(EscapeError::EscapeOnlyChar),
-            '\r' => Err(if chars.clone().next() == Some('\n') {
-                EscapeError::EscapeOnlyChar
-            } else {
-                EscapeError::BareCarriageReturn
-            }),
+            '\t' | '\r' | '\n' => Err(EscapeError::EscapeOnlyChar),
```

Comment:
If `\r` is prohibited at load time, them it can be removed from this list to avoid extra errors.